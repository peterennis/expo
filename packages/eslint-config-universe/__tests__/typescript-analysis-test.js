const path = require('path');

const getBaseConfig = require('./tools/getBaseConfig');
const lintAsync = require('./tools/lintAsync');

const configFile = path.resolve(__dirname, '../shared/typescript-analysis.js');

const alteredBaseConfig = {
  ...getBaseConfig(),
  parserOptions: {
    project: './__tests__/fixtures/tsconfig.json',
  },
};

it(`lints`, async () => {
  let report = await lintAsync(
    {
      baseConfig: alteredBaseConfig,
      configFile,
      fix: true,
      ignore: false,
      useEslintrc: false,
    },
    ['__tests__/fixtures/*typescript-analysis*']
  );
  let { results } = report;
  for (let result of results) {
    let relativeFilePath = path.relative(__dirname, result.filePath);
    delete result.filePath;
    expect(result).toMatchSnapshot(relativeFilePath);
  }
}, 20000);
