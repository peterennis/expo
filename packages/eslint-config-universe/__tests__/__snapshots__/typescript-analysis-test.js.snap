// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lints: fixtures/typescript-analysis-00.ts 1`] = `
Object {
  "errorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": Array [],
  "output": "type T = {
  a?: {
    b?: {
      c: string;
      method?: () => void;
    };
  };
};

export function myFunc(foo: T | null) {
  return foo?.a?.b?.c;
}",
  "warningCount": 0,
}
`;

exports[`lints: fixtures/typescript-analysis-01.ts 1`] = `
Object {
  "errorCount": 1,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": Array [
    Object {
      "column": 1,
      "endColumn": 2,
      "endLine": 6,
      "line": 4,
      "message": "For-in loops over arrays are forbidden. Use for-of or array.forEach instead.",
      "messageId": "forInViolation",
      "nodeType": "ForInStatement",
      "ruleId": "@typescript-eslint/no-for-in-array",
      "severity": 2,
    },
  ],
  "source": "export const arr = [1, 2, 3];

let sum = '';
for (const b in arr) {
  sum += b;
}
export { sum };",
  "warningCount": 0,
}
`;

exports[`lints: fixtures/typescript-analysis-02.ts 1`] = `
Object {
  "errorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": Array [],
  "output": "export function getBlah(): string | null {
  const a: string | null = Math.random() ? '' : null;
  const b: string | null = Math.random() ? '' : null;
  return a ?? b;
}",
  "warningCount": 0,
}
`;

exports[`lints: fixtures/typescript-analysis-03.ts 1`] = `
Object {
  "errorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": Array [
    Object {
      "column": 9,
      "endColumn": 10,
      "endLine": 2,
      "line": 2,
      "message": "Expected an error object to be thrown.",
      "messageId": "object",
      "nodeType": "Literal",
      "ruleId": "@typescript-eslint/no-throw-literal",
      "severity": 1,
    },
  ],
  "source": "export function wat() {
  throw 3;
}",
  "warningCount": 1,
}
`;
